java/src/main/native/src/map_lookup.cu

```diff
@@ -124,27 +127,59 @@ get_gather_map_for_map_values(column_view const &input, string_scalar &lookup_ke
   return gather_map;
 }
 
-} // namespace
-
-namespace jni {
-std::unique_ptr<column> map_lookup(column_view const &map_column, string_scalar lookup_key,
-                                   bool has_nulls, rmm::cuda_stream_view stream,
-                                   rmm::mr::device_memory_resource *mr) {
-  // Defensive checks.
+/**
+ * @brief a defensive check for the map column that is going to be processed
+ */
+void map_check(column_view const &map_column, rmm::cuda_stream_view stream) {
   CUDF_EXPECTS(map_column.type().id() == type_id::LIST, "Expected LIST<STRUCT<key,value>>.");
 
   lists_column_view lcv{map_column};
-  auto structs_column = lcv.get_sliced_child(stream);
+  column_view structs_column = lcv.get_sliced_child(stream);
 
   CUDF_EXPECTS(structs_column.type().id() == type_id::STRUCT, "Expected LIST<STRUCT<key,value>>.");
 
-  structs_column_view scv{structs_column};
   CUDF_EXPECTS(structs_column.num_children() == 2, "Expected LIST<STRUCT<key,value>>.");
   CUDF_EXPECTS(structs_column.child(0).type().id() == type_id::STRING,
                "Expected LIST<STRUCT<key,value>>.");
   CUDF_EXPECTS(structs_column.child(1).type().id() == type_id::STRING,
                "Expected LIST<STRUCT<key,value>>.");
+}
+
+} // namespace
 
+namespace jni {
+
+std::unique_ptr<scalar> map_contains(column_view const &map_column, string_scalar lookup_key,
+                                     bool has_nulls, rmm::cuda_stream_view stream,
+                                     rmm::mr::device_memory_resource *mr) {
+  // Defensive checks.
+  map_check(map_column, stream);
+
+  lists_column_view lcv(map_column);
+  structs_column_view scv(lcv.child());
+
+  std::vector<column_view> children;
+  children.push_back(lcv.offsets());
+  children.push_back(scv.child(0));
+
+  column_view list_of_keys(map_column.type(), map_column.size(),
+    nullptr, map_column.null_mask(), map_column.null_count(), 0, children);
+  auto contains_column  = lists::contains(list_of_keys, lookup_key);
+  // null will skipped in all-aggregation, so mask all nulls with 0.
+  contains_column->set_null_mask(rmm::device_buffer{0}, 0);
```

Comment:
This makes the assumption that the value memory underlying a null has been initialized to a specific value, and I don't think it is safe to assume the `list::contains` implementation will always guarantee that.  It may today, but how will we know it always will guarantee it unless it is part of the formal contract of that function?  I think this needs to call `replace_nulls` instead here.